data-tx/src/main/java/io/micronaut/transaction/sync/SynchronousFromReactiveTransactionManager.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    @Override
+    public T getConnection() {
+        throw noSupported();
+    }
+
+    @Override
+    public boolean hasConnection() {
+        throw noSupported();
+    }
+
+    @Override
+    public <R> R execute(TransactionDefinition definition, TransactionCallback<T, R> callback) {
+        return TransactionSynchronizationManager.withGuardedState(() -> {
+            TransactionSynchronizationManager.TransactionSynchronizationState state = TransactionSynchronizationManager.getOrCreateState();
+            Context previousContext = (Context) TransactionSynchronizationManager.unbindResourceIfPossible(ContextView.class);
+            Mono<R> result = reactiveTransactionOperations.withTransactionMono(definition, status -> {
+                return Mono.deferContextual(contextView -> {
+                    return Mono.justOrEmpty(TransactionSynchronizationManager.withState(state, () -> {
+                        TransactionSynchronizationManager.bindResource(ContextView.class, contextView);
+                        return callback.apply(new DefaultTransactionStatus<>(status));
+                    }));
+                }).doAfterTerminate(() -> {
+                    TransactionSynchronizationManager.withState(state, () -> {
+                        TransactionSynchronizationManager.unbindResourceIfPossible(ContextView.class);
+                        return null;
+                    });
+                }).subscribeOn(scheduler);
+            });
+            if (previousContext != null) {
+                result = result.contextWrite(previousContext);
+            }
+            return result.onErrorMap(e -> {
+                if (e instanceof UndeclaredThrowableException) {
+                    return e.getCause();
+                }
+                return e;
+            }).block();
+        });
+    }
+
+    @Override
+    public <R> R executeRead(TransactionCallback<T, R> callback) {
+        return execute(TransactionDefinition.READ_ONLY, callback);
+    }
+
+    @Override
+    public <R> R executeWrite(TransactionCallback<T, R> callback) {
+        return execute(TransactionDefinition.DEFAULT, callback);
+    }
+
+    @NotNull
+    private IllegalStateException noSupported() {
+        return new IllegalStateException("This synchronous transaction manager is implemented using blocking of the reactive transaction manager " +
+                "and only supports 'execute', 'executeRead' and 'executeWrite' methods.");
+    }
+
+    private static final class DefaultTransactionStatus<T> implements TransactionStatus<T> {
+
+        private final ReactiveTransactionStatus<T> transactionStatus;
+
+        private DefaultTransactionStatus(ReactiveTransactionStatus<T> transactionStatus) {
+            this.transactionStatus = transactionStatus;
+        }
+
+        @Override
+        public Object createSavepoint() throws TransactionException {
+            throw new IllegalStateException("Not supported!");
```

Comment:
call `noSupported` in the outer class to improve this error for this method all others here